web/src/main/java/se/inera/intyg/webcert/web/service/facade/list/ListDecorator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2022 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.webcert.web.service.facade.list;
+
+import se.inera.intyg.webcert.web.web.controller.api.dto.ListIntygEntry;
+
+import java.util.List;
+
+public interface ListDecorator {
+    List<ListIntygEntry> decorateWithCertificateTypeName(List<ListIntygEntry> list);
```

Comment:
Is it really decoration when it returns a new list? Do you need to have a return value because the method is changing the items that are sent in as parameters.

Reply:
You can simplify the implementation if you change this.